skimage/segmentation/tests/test_join.py

```diff
@@ -87,7 +87,7 @@ def test_relabel_sequential_offset5_with0():
 
 
 def test_relabel_sequential_dtype():
-    ar = np.array([1, 1, 5, 5, 8, 99, 42, 0], dtype=float)
+    ar = np.array([1, 1, 5, 5, 8, 99, 42, 0], dtype=np.uint8)
```

Comment:
Should we raise an error if input dtype is not integer?

Reply:
We don't have a strict policy on this. Often, if the docstring specifies the dtype, we consider that sufficient warning. In this case, the Cython code will fail. Happy to go either way, but I worry about functions that have a big chunk of validity checking at the top. It's a distraction.

Reply:
An error raised by us is probably easier to understand than a failure of Cython... I'd be :+1: for raising on float.